nemo/collections/asr/parts/mixins/interctc_mixin.py

```diff
@@ -189,29 +206,33 @@ def add_interctc_losses(
         """
         if not self.is_interctc_enabled() or not AccessMixin.is_access_enabled():
             return loss_value, {}
-        metrics = {f"{log_prefix}final_ctc_loss": loss_value}
+        metrics = {f"{log_prefix}final_loss": loss_value}
         captured_tensors = self.get_captured_interctc_tensors()
 
+        loss_value *= self._interctc_params['main_loss_weight']
+
         for layer_idx, intermediate_result, loss_weight in zip(
-            self._apply_at_layers, captured_tensors, self._intermediate_loss_weights
+            self._interctc_params['apply_at_layers'],
+            captured_tensors,
+            self._interctc_params['intermediate_loss_weights'],
         ):
-            inter_loss_value = self.loss(
+            inter_loss_value = self._interctc_params['loss'](
                 log_probs=intermediate_result[0],
                 targets=transcript,
                 target_lengths=transcript_len,
                 input_lengths=intermediate_result[1],
             )
-            metrics[f"{log_prefix}inter_ctc_loss_l{layer_idx}"] = loss_value.detach()
+            metrics[f"{log_prefix}inter_ctc_loss_l{layer_idx}"] = inter_loss_value.detach()
             loss_value += inter_loss_value * loss_weight
             if compute_wer:
-                self._wer.update(
+                self._interctc_params['wer'].update(
```

Comment:
Why not use self._wer ? You can check and set self._wer or self.wer (or better yet, add alias of self.wer = self._wer as a property for CTC models.